cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/DefaultApplicationUpdateContext.java

```diff
@@ -0,0 +1,164 @@
+import io.cdap.cdap.common.io.CaseInsensitiveEnumTypeAdapterFactory;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactDescriptor;
+import io.cdap.cdap.internal.app.runtime.artifact.ArtifactRepository;
+import io.cdap.cdap.proto.artifact.ArtifactSortOrder;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.id.NamespaceId;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Default implementation of {@link ApplicationUpdateContext}.
+ *
+ * Used during update of an Application config via provided helper methods used during an update action like
+ * upgrade/downgrade.
+ */
+public class DefaultApplicationUpdateContext implements ApplicationUpdateContext {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DefaultApplicationUpdateContext.class);
+  private static final Gson GSON = new GsonBuilder().
+      registerTypeAdapterFactory(new CaseInsensitiveEnumTypeAdapterFactory()).create();
+
+  private final ArtifactId applicationArtifactId;
+  private String configString;
+  private List<ApplicationConfigUpdateAction> updateActions;
+  private final ArtifactRepository artifactRepository;
+  private final ApplicationId applicationId;
+  private final NamespaceId namespaceId;
+
+  public DefaultApplicationUpdateContext(NamespaceId namespaceId, ApplicationId applicationId,
+                                         ArtifactId applicationArtifactId, ArtifactRepository artifactRepository,
+                                         String configString, List<ApplicationConfigUpdateAction> updateActions) {
+    this.namespaceId = namespaceId;
+    this.applicationId = applicationId;
+    this.artifactRepository = artifactRepository;
+    this.applicationArtifactId = applicationArtifactId;
+    this.configString = configString;
+    this.updateActions = updateActions;
+  }
+
+  @Override
+  public List<ApplicationConfigUpdateAction> getUpdateActions() {
+    return updateActions;
+  }
+
+  @Override
+  public <C extends Config> C getConfig(Type configType) {
+    C appConfig;
+
+    if (configString.isEmpty()) {
+      try {
+        appConfig = ((Class<C>) configType).newInstance();
+      } catch (Exception e) {
+        throw new IllegalArgumentException("Issue in creating config class of type " + configType.getTypeName(), e);
+      }
+    } else {
+      try {
+        appConfig = GSON.fromJson(configString, configType);
+      } catch (JsonSyntaxException e) {
+        throw new IllegalArgumentException("Invalid JSON application configuration was provided. Please check the"
+                                           + " syntax.", e);
+      }
+    }
+    return appConfig;
+  }
+
+  @Override
+  public String getConfigAsString() {
+    return configString;
+  }
+
+  // Returns plugin artifacts using given filters in ascending order.
```

Comment:
When writing doc for a method, please with the method javadoc syntax instead of line comment.